pkg/antctl/raw/multicluster/common/common.go

```diff
@@ -0,0 +1,210 @@
+	if err != nil {
+		return nil, err
+	}
+	restConfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restConfigTmpl)
+
+	k8sClient, err := client.New(restConfigTmpl, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sClient, nil
+}
+
+func CreateClusterClaim(cmd *cobra.Command, k8sClient client.Client, namespace string, clusterset string, clusterID string, createdRes *[]*unstructured.Unstructured) error {
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+	var createErr error
+	if createErr = k8sClient.Create(context.TODO(), newClusterClaim(clusterID, namespace, false)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimID, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, namespace)
+		*createdRes = append(*createdRes, &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "multicluster.crd.antrea.io/v1alpha2",
+				"kind":       "ClusterClaim",
+				"metadata": map[string]interface{}{
+					"name":      multiclusterv1alpha2.WellKnownClusterClaimID,
+					"namespace": namespace,
+				},
+			},
+		})
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, namespace)
+	if createErr = k8sClient.Create(context.TODO(), newClusterClaim(clusterset, namespace, true)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, namespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, namespace)
+		*createdRes = append(*createdRes, &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "multicluster.crd.antrea.io/v1alpha2",
+				"kind":       "ClusterClaim",
+				"metadata": map[string]interface{}{
+					"name":      multiclusterv1alpha2.WellKnownClusterClaimClusterSet,
+					"namespace": namespace,
+				},
+			},
+		})
+	}
+
+	return nil
+}
+
+func CreateClusterSet(cmd *cobra.Command, k8sClient client.Client, namespace string, clusterset string,
```

Comment:
I feel you can pass the `joinOpts` directly instead of a long parameter list.

Reply:
I used to pass the `joinOpts` but got an error of circle import. So maybe we have to pass the value.